src/Microsoft.AspNetCore.Mvc.Core/Internal/FileContentResultExecutor.cs

```diff
@@ -15,15 +17,54 @@ public FileContentResultExecutor(ILoggerFactory loggerFactory)
 
         public Task ExecuteAsync(ActionContext context, FileContentResult result)
         {
+            RangeItemHeaderValue range = new RangeItemHeaderValue(0, 0);
+            long rangeLength = default(long);
+            if (result.EnableRangeProcessing)
+            {
+                range = SetContentRangeAndStatusCode(
+                    context,
+                    result.LastModified,
+                    result.EntityTag,
+                    result.FileContents.Length);
+
+                rangeLength = (range == null) ? 0 : SetRangeHeaders(context, result, range);
+            }
+
             SetHeadersAndLog(context, result);
-            return WriteFileAsync(context, result);
+            return WriteFileAsync(context, result, range, rangeLength);
         }
 
-        private static Task WriteFileAsync(ActionContext context, FileContentResult result)
+        private static Task WriteFileAsync(ActionContext context, FileContentResult result, RangeItemHeaderValue range, long rangeLength)
         {
             var response = context.HttpContext.Response;
+            var outputStream = response.Body;
+
+            if (!result.EnableRangeProcessing)
+            {
+                return response.Body.WriteAsync(result.FileContents, offset: 0, count: result.FileContents.Length);
+            }
+
+            else if (range == null || rangeLength == 0)
```

Comment:
if no range was specified (as opposed to a range length of 0) then you should send the whole body.